tinygrad/codegen/kernel.py

```diff
@@ -390,16 +394,21 @@ def fix(needed, ax):
                 self.apply_opt(Opt(OptOps.LOCAL, s0, upc))
                 break
 
-        # alias buffer
-        alias_pattern = [0]*(self.global_dims+(self.local_dims-len(tc.threads))) + [2]*(len(tc.threads)) + [0]*(self.shape_len-self.upcasted-self.first_reduce) + [1,1] + [3]*(self.upcasted-2)  # noqa: E501
-        self.alias_buffer(buf0, alias_pattern)
-        self.alias_buffer(buf1, alias_pattern)
+        # delay alias_buffer until all optimizations are completed
+        self.tensor_core_bufs = [buf0, buf1]
         return True
     return False
 
-  def apply_opt(self, opt:Opt):
+  def apply_opt(self, opt:Opt, append_opt:bool=True):
```

Comment:
Hmm, so if I construct this with apply_tensor_cores it's never appended? I don't like this as an option, it should always append and be able to reconstruct.

Aka, the Linearizer state should always be able to be recreated with the ast + applied_opts

Reply:
Whoops good catch - this wasn't intended.  Will push a fix shortly.

Reply:
Can you add an explicit test for this reconstruction?

Reply:
Added this test in `test_tensor_cores`